package.json

```diff
@@ -83,6 +83,7 @@
     "license-checker": "^25.0.1",
     "mime": "^3.0.0",
     "ncp": "^2.0.0",
+    "nock": "^13.2.7",
```

Comment:
would be also nice if we could drop the nock dependency. I'd do it the following way:

1. when an env var is set, use a localhost instead of the official ado host
2. spawn a http server inside our tests and return the relevant responses

Reply:
@mxschmitt i'm have some situation with this step
i'm using [TestServer](https://github.com/microsoft/playwright/tree/main/utils/testserver) to write ADO mock server where i'm defining all i needed routs with response
I took one first and faced some difficulties
i'm connect test server as pw config
```ts
webServer: {
  command: 'node ${JSON.stringify(SIMPLE_SERVER_PATH)} ${port}',
  port: ${port},
},
```
and when test runned it failed because last request from reporter does not intercepting with test server - test server will be destroyed with pw test-runner (i think so), and last request from azure reporter onEnd hook does not be responding
i solved it using `spawn` from `child_process` package:
1. spawn node process with server
2. run test with 'runInlineTest' method 
3. kill spawned process with `child.kill('SIGINT')` command
4. check test results

this variant working
else maybe you have some idea?

Reply:
We have an inline fixture for our server see here: https://github.com/microsoft/playwright/blob/d12afa7ff399c8a9479394bcb1fc492d192e8173/tests/playwright-test/web-server.spec.ts#L434-L437

If you want to server static assets, I recommend to put them into `tests/assets/playwright-test/reporters/azure` and use then `server.serveFile` inside your actual test.

I think using this makes the other questions irrelevant? Feel free to ping me on Slack as well if you have further questions: https://aka.ms/playwright-slack